packages/cli/src/utils/prompt.ts

```diff
@@ -2,39 +2,64 @@ import isEmpty from 'lodash.isempty';
 import Inquirer from 'inquirer';
 import Truffle from '../models/initializer/truffle/Truffle';
 
-interface PromptParams {
+interface Args {
   args?: {};
+  opts?: {};
   props?: {};
   defaults?: {};
 }
 
+interface GenericObject {
+  [key: string]: any;
+}
+
 // TS-TODO: Define a more accurate return type as soon as we know the final structure of it
-export async function promptForArgumentsIfNeeded({ args, defaults, props }: PromptParams): Promise<any> {
-  const questions = Object.keys(args)
+export async function promptIfNeeded({ args = {}, opts = {}, defaults, props }: Args): Promise<any> {
+  const argsQuestions = Object.keys(args)
     .filter((argName) => !args[argName] || isEmpty(args[argName]))
     .map((argName) => promptFor(argName, defaults, props));
 
-  return { ...args, ...(await Inquirer.prompt(questions)) };
+  const optsQuestions = Object.keys(opts)
+    .filter((optName) => !opts[optName])
```

Comment:
What happens if the user explicitly sets `false` to an opt via command line? Shouldn't we be checking for undefined instead?

Reply:
`false` would be passed as a string, so there shouldn't be any problem. Anyway, It's a good practice to be more explicit in these cases as it will be clearer to understand whether it should filter the option or not. Good gatch :slightly_smiling_face: